plugin/PDFDocumentPage.qml

```diff
@@ -55,67 +62,41 @@ DocumentPage {
     }
     Connections {
         target: view
-        onPageSizesReady: if (rememberPositionConfig.value) {
-            if (!_settings) {
-                _settings = new PDFStorage.Settings(pdfDocument.source)
-            }
-            var last = _settings.getLastPage()
-            if (last[3] > 0) {
-                view.itemWidth = last[3]
-                view.adjust()
+        onPageSizesReady: {
+            if (rememberPositionConfig.value) {
+                if (!_settings) {
+                    _settings = new PDFStorage.Settings(pdfDocument.source)
+                }
+                var last = _settings.getLastPage()
+                if (last[3] > 0) {
+                    view.itemWidth = last[3]
+                    view.adjust()
+                }
+                view.goToPage( last[0] - 1, last[1], last[2] )
             }
-            view.goToPage( last[0] - 1, last[1], last[2] )
         }
     }
 
+    Binding {
+        target: base
+        property: "forwardNavigation"
+        value: false
+        when: (pdfDocument.failure || pdfDocument.locked)
+    }
+
+    Loader {
+        parent: base
```

Comment:
Because DocumentPage has default property that otherwise parents to container item.